.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,47 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.241.1/containers/typescript-node
+{
+	"name": "Node.js & TypeScript",
+	"build": {
+		"dockerfile": "Dockerfile",
+		// Update 'VARIANT' to pick a Node version: 18, 16, 14.
+		// Append -bullseye or -buster to pin to an OS version.
+		// Use -bullseye variants on local on arm64/Apple Silicon.
+		"args": {
+			"VARIANT": "18"
+		}
+	},
+	// Configure tool-specific properties.
+	"customizations": {
+		// Configure properties specific to VS Code.
+		"vscode": {
+			// Add the IDs of extensions you want installed when the container is created.
+			"extensions": [
+				"dbaeumer.vscode-eslint",
+				"unifiedjs.vscode-mdx",
+				"ms-azuretools.vscode-docker",
+				"github.vscode-pull-request-github",
+				"eamodio.gitlens",
+				"davidanson.vscode-markdownlint",
+				"christian-kohler.npm-intellisense",
+				"orta.vscode-jest",
+				"firsttris.vscode-jest-runner",

```

Comment:
I'm not quite sure if these are working correctly - when I start the devcontainer I see the status bar at the bottom has a red ! for Jest, and the terminal says that it cannot find the executable. Is there a specific command I can run to test if it is actually working? Alternatively, what should these two plugins give me in the UI?

Reply:
yeah, looks like there's an issue with the jest setup:

  ● Test suite failed to run 

    ReferenceError: module is not defined in ES module scope
    This file is being treated as an ES module because it has a '.js' file extension and '/workspaces/frontend/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.

    > 1 | module.exports = 'test-file-stub';


 Switching it to `export default 'testFileStub'` and setting it to typescript worked on my end, though I think the issue lies with the configuration more than anything. Also might be I messed up my container whilst setting it up >_>

Reply:
So, Jest config was partly broken (I set it up for v27 and they changed how a lot of things worked for v28 and v29), partly had the issue Dantarian mentioned in the Dockerfile, and partly had the auto-run running before everything was ready. I've fixed the config and changed the autorun settings so this should be fine now.